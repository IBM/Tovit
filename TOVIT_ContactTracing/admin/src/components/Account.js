/* globals alert */
import React, { useState } from 'react'
import { useTranslation } from 'react-i18next'
import { updatePassword } from '../helpers/api'

export function Account () {
  const [currentPassword, setCurrentPassword] = useState('')
  const [newPassword, setNewPassword] = useState('')
  const { t } = useTranslation()

  const onSubmitNewPassword = async (event) => {
    event.preventDefault()
    const passwordRegex = /^[\w!@#$%^&*]{8,}$/
    if (passwordRegex.test(newPassword)) {
      const updatePasswordSuccess = await updatePassword(currentPassword, newPassword)
      if (!updatePasswordSuccess) {
        alert(t('account_current_password_incorrect'))
      } else {
        alert(t('account_update_password_success'))
        setCurrentPassword('')
        setNewPassword('')
      }
    } else {
      alert(t('account_new_password_invalid'))
    }
  }

  const onchangeCurrentPassword = (event) => {
    setCurrentPassword(event.target.value)
  }

  const onchangeNewPassword = (event) => {
    setNewPassword(event.target.value)
  }

  return (
    <div>
      <div class='d-flex justify-content-between flex-wrap flex-md-nowrap align-items-left pt-3 pb-2 mb-3 border-bottom'>
        <h1 class='h2'>{t('account_title')}</h1>
      </div>
      <form class='form-signin' onSubmit={onSubmitNewPassword}>
        <h2 class='h3 mb-3 font-weight-normal'>{t('account_update_password_title')}</h2>
        <p>{t('account_password_instructions')}</p>
        <label for='current-password' class='sr-only'>{t('account_current_password')}:</label>
        <input value={currentPassword} onChange={onchangeCurrentPassword} type='password' id='username' name='current-password' placeholder={t('account_current_password')} class='form-control' />
        <label for='new-password' class='sr-only'>{t('account_new_password')}:</label>
        <input value={newPassword} onChange={onchangeNewPassword} type='password' id='username' name='new-password' placeholder={t('account_new_password')} class='form-control' />
        <button class='btn btn-lg btn-warning btn-block mt-3' type='submit'>{t('account_update_password_submit')}</button>
      </form>
    </div>
  )
}
