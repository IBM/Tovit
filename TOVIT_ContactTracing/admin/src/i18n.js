import i18n from 'i18next'
import { initReactI18next } from 'react-i18next'
import LanguageDetector from 'i18next-browser-languagedetector'
import en from './translations/en'

// the translations
// (tip move them in a JSON file and import them)
const resources = {
  en
}

i18n
  .use(LanguageDetector)
  .use(initReactI18next) // passes i18n down to react-i18next
  .init({
    resources,
    fallbackLng: 'en',
    keySeparator: false, // we do not use keys in form messages.welcome
    interpolation: {
      escapeValue: false // react already safes from xss
    },
    detection: {
      order: ['querystring', 'cookie', 'localStorage', 'navigator', 'htmlTag', 'path', 'subdomain'],

      // keys or params to lookup language from
      lookupQuerystring: 'lng',
      lookupCookie: 'i18next',
      lookupLocalStorage: 'i18nextLng',
      lookupFromPathIndex: 0,
      lookupFromSubdomainIndex: 0,

      // cache user language on
      caches: ['localStorage', 'cookie'],
      excludeCacheFor: ['cimode'], // languages to not persist (cookie, localStorage)

      // optional expire and domain for set cookie
      cookieMinutes: 10,
      cookieDomain: 'myDomain',

      // optional htmlTag with lang attribute, the default is:
      htmlTag: document.documentElement,

      // only detect languages that are in the whitelist
      checkWhitelist: true
    }
  })

export default i18n
