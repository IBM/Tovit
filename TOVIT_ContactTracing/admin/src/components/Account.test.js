/* globals test, expect */
import React from 'react'
import { Account } from './Account'
import TestRenderer from 'react-test-renderer'
import { I18nextProvider } from 'react-i18next'
import i18n from '../i18n'

test('Account component', () => {
  const component = TestRenderer.create(
    <I18nextProvider i18n={i18n}>
      <Account />
    </I18nextProvider>)
  expect(component).toMatchSnapshot()
})
