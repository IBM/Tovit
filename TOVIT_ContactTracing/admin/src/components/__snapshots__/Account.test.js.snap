// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Account component 1`] = `
<div>
  <div
    class="d-flex justify-content-between flex-wrap flex-md-nowrap align-items-left pt-3 pb-2 mb-3 border-bottom"
  >
    <h1
      class="h2"
    >
      Account
    </h1>
  </div>
  <form
    class="form-signin"
    onSubmit={[Function]}
  >
    <h2
      class="h3 mb-3 font-weight-normal"
    >
      Update password
    </h2>
    <p>
      Passwords must be at least 8 characters long.
    </p>
    <label
      class="sr-only"
      for="current-password"
    >
      Current password
      :
    </label>
    <input
      class="form-control"
      id="username"
      name="current-password"
      onChange={[Function]}
      placeholder="Current password"
      type="password"
      value=""
    />
    <label
      class="sr-only"
      for="new-password"
    >
      New password
      :
    </label>
    <input
      class="form-control"
      id="username"
      name="new-password"
      onChange={[Function]}
      placeholder="New password"
      type="password"
      value=""
    />
    <button
      class="btn btn-lg btn-warning btn-block mt-3"
      type="submit"
    >
      Update
    </button>
  </form>
</div>
`;
